### PR TITLE
Adding my Random Menu Generator project

### PR DESCRIPTION
**Pull request by shannachau**
Completed at the dessert level

### CODE REVIEW

File: randommenu_platinum.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Random Menu Generator
+
+# create empty arrays
+adjective = []
+cook_method = []
+food_item = []
```
Comment (khjrtbrg) [inline randommenu_platinum.rb:6]: It might be good to have these variable names be plural, to indicate that there may be more than one item in each.

File: randommenu_platinum.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+# Get user input for how many item menus
+puts "Hey! Let's generate a random menu for you.\nHow many items would you like on your menu?"
+list_num = gets.chomp
+
+# Error message for invalid number
+until list_num.to_i.to_s == list_num && list_num.to_i > 0
```
Comment (khjrtbrg) [inline randommenu_platinum.rb:13]: Nice quality control here. Converting `list_num` multiple times here is a little hard to follow, though. If you're trying to assert that the number your user has entered is an actual number and it's greater than zero, you should be able to get away with just writing:

``` ruby
until list_num.to_i > 0
```

(Strings that can't convert directly to numbers are converted to zero and you're making sure their input is greater than zero already.)

File: randommenu_platinum.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  puts "That's NOT a valid number! Choose again."
+  list_num = gets.chomp
+end
+
+# Convert user input from string to integer
+list_num = list_num.to_i
```
Comment (khjrtbrg) [inline randommenu_platinum.rb:19]: You could do this on the same line as your `gets.chomp` and then just check that it's greater than zero on line 13.

File: randommenu_platinum.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+# Prompt user to enter list of adjectives that equal the number of menu items
+puts "\nGreat! #{list_num} items it is! \nLet's start by entering in #{list_num} adjectives!"
+
+# Adds each adjective user inputs into the array
+i = 1
+until i > list_num
```
Comment (khjrtbrg) [inline randommenu_platinum.rb:26]: This totally works, but you could also use `times`:

``` ruby
list_num.times do
```

File: randommenu_platinum.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  user_adjective = gets.chomp
+  adjective.push(user_adjective)
```
Comment (khjrtbrg) [inline randommenu_platinum.rb:28]: What happens if someone hits enter without typing anything or if someone enters a number?

File: randommenu_platinum.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  food_item.push(user_food_item)
+  i += 1
+end
+
+puts "\nHere's your custom menu! Enjoy!"
+
+# Shuffles/randomizes order of items in each array
+adjective = adjective.shuffle
+cook_method = cook_method.shuffle
+food_item = food_item.shuffle
```
Comment (khjrtbrg) [inline randommenu_platinum.rb:59]: It might be good to assign these to new variables that describe the array's new state, rather than overwriting the existing variable. For instance, `shuffled_adjectives` or `random_adjectives`.

File: randommenu_platinum.rb [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+# Outputs menu with no repeated items
```
Comment (khjrtbrg) [inline randommenu_platinum.rb:61]: Awesome way of handling no repeated items, by the way.

File: randommenu_platinum.rb [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+menu = 1
+until menu == list_num + 1
```
Comment (khjrtbrg) [inline randommenu_platinum.rb:63]: It might be a bit easier to read if you wrote `menu > list_num`. It took me a second to realize why you were comparing `menu` to `list_num + 1`. Alternatively, `times` would be a great option here (it has a built in variable keeping track of the current iteration), or even calling `each_with_index` on one array and use the current iteration's index to look up the values in the other arrays.

**Comment by khjrtbrg** [comment]
Good job @shannachau! I liked how you shuffled the arrays before beginning to print out the menu, since it let you loop through the arrays in a straightforward manner while also guaranteeing no repeats. You might want to consider handling edge cases like users entering nothing when you're collecting your word lists, but overall it looks great! Let me know if you have any questions or if you want me to follow up on anything. :+1:

File: randommenu_platinum.rb (hunk 2 above)
Comment (shannachau) [inline randommenu_platinum.rb:13]: Great tip!! Looking back, it was definitely a roundabout way to do it.

File: randommenu_platinum.rb (hunk 6 above)
Comment (shannachau) [inline randommenu_platinum.rb:59]: Didn't think about this before. I like the variables you suggested. Definitely makes the code clearer.

**Comment by shannachau** [comment]
Thanks so much for taking the time to do this code review!! There are a few things I can definitely do to improve clarity for my code. All of your suggestions were super helpful!!
